### ISSUE TITLE
Kernel Oops on zpool import

### ISSUE THREAD

**Issue by mikedilger** [zfsonlinux/zfs#2457]
After a system hang and power cycle, zfs import causes a kernel Oops in [zfs].
I'm on zfs 0.6.3, gentoo, x86_64.  I tried going back to 0.6.2, and it still happens.  I tried upgrading from kernel 3.12.x to 3.15.3 and there is no change (as expected).

Below is the dmesg with stack information.

[  858.551047] BUG: unable to handle kernel paging request at ffffffffa0c3ed48
[  858.551061] IP: [<ffffffffa0b85801>] zio_buf_alloc+0x11/0x20 [zfs]
[  858.551083] PGD 1e11067 PUD 1e12063 PMD 0 
[  858.551090] Oops: 0000 [#1] PREEMPT SMP 
[  858.551098] Modules linked in: zfs(PO) zunicode(PO) zavl(PO) zcommon(PO) znvpair(PO) spl(O) nvidia(PO) ipv6 iTCO_wdt coretemp crc32c_intel lpc_ich mfd_core
[  858.551121] CPU: 1 PID: 16209 Comm: spl_system_task Tainted: P           O  3.15.3 #3
[  858.551125] Hardware name: Gigabyte Technology Co., Ltd. H57M-USB3/H57M-USB3, BIOS F1 01/06/2010
[  858.551127] task: ffff8800a1f62400 ti: ffff8800a1dcc000 task.ti: ffff8800a1dcc000
[  858.551130] RIP: 0010:[<ffffffffa0b85801>]  [<ffffffffa0b85801>] zio_buf_alloc+0x11/0x20 [zfs]
[  858.551144] RSP: 0018:ffff8800a1dcfa38  EFLAGS: 00010213
[  858.551147] RAX: 000000004b400000 RBX: 0000000001f01c00 RCX: ffffffffa0bbbe20
[  858.551149] RDX: 0000000100088649 RSI: 0000000000000230 RDI: 000000000000f80d
[  858.551152] RBP: ffff8800a1dcfa38 R08: 566e8ed20eb026cb R09: 566e8ed20eb026cb
[  858.551154] R10: 000000000000001f R11: 0000000000000000 R12: 0000000000000001
[  858.551156] R13: ffff880074add130 R14: ffff880074add168 R15: 000000000000f80d
[  858.551159] FS:  0000000000000000(0000) GS:ffff880127c40000(0000) knlGS:0000000000000000
[  858.551162] CS:  0010 DS: 0000 ES: 0000 CR0: 000000008005003b
[  858.551164] CR2: ffffffffa0c3ed48 CR3: 0000000001e10000 CR4: 00000000000007e0
[  858.551166] Stack:
[  858.551168]  ffff8800a1dcfa70 ffffffffa0af30dd ffff88010d835a80 ffff880074add130
[  858.551173]  0000000001f01c00 0000000000000001 000000000000f80d ffff8800a1dcfaa0
[  858.551182]  ffffffffa0af5917 ffffc9000bdeff40 0000000000000000 ffff8800a1dcfb6c
[  858.551186] Call Trace:
[  858.551196]  [<ffffffffa0af30dd>] arc_space_return+0xbcd/0x3330 [zfs]
[  858.551213]  [<ffffffffa0af5917>] arc_buf_alloc+0xd7/0x100 [zfs]
[  858.551226]  [<ffffffffa0af6112>] arc_read+0x362/0x8e0 [zfs]
[  858.551238]  [<ffffffffa0b8580e>] ? zio_buf_alloc+0x1e/0x20 [zfs]
[  858.551247]  [<ffffffffa0af30ed>] ? arc_space_return+0xbdd/0x3330 [zfs]
[  858.551263]  [<ffffffffa0b0c61d>] dmu_objset_is_receiving+0xbd/0xe30 [zfs]
[  858.551280]  [<ffffffffa0b0c68e>] dmu_objset_is_receiving+0x12e/0xe30 [zfs]
[  858.551301]  [<ffffffffa0b0ce54>] dmu_objset_is_receiving+0x8f4/0xe30 [zfs]
[  858.551322]  [<ffffffffa0b0cd6b>] dmu_objset_is_receiving+0x80b/0xe30 [zfs]
[  858.551338]  [<ffffffffa0b0d6b5>] traverse_pool+0x2d5/0x380 [zfs]
[  858.551354]  [<ffffffffa0b0cf61>] dmu_objset_is_receiving+0xa01/0xe30 [zfs]
[  858.551370]  [<ffffffffa0b0d5fe>] traverse_pool+0x21e/0x380 [zfs]
[  858.551386]  [<ffffffffa0b0c6e0>] ? dmu_objset_is_receiving+0x180/0xe30 [zfs]
[  858.551392]  [<ffffffffa0015dd8>] taskq_cancel_id+0x2c8/0x470 [spl]
[  858.551398]  [<ffffffff810868c0>] ? wake_up_state+0x10/0x10
[  858.551404]  [<ffffffffa0015c30>] ? taskq_cancel_id+0x120/0x470 [spl]
[  858.551410]  [<ffffffff81077a04>] kthread+0xc4/0xe0
[  858.551417]  [<ffffffff81077940>] ? kthread_create_on_node+0x170/0x170
[  858.551426]  [<ffffffff818fd4bc>] ret_from_fork+0x7c/0xb0
[  858.551432]  [<ffffffff81077940>] ? kthread_create_on_node+0x170/0x170
[  858.551436] Code: e8 95 57 fa ff 5b 41 5c 41 5d 41 5e 5d c3 66 66 66 2e 0f 1f 84 00 00 00 00 00 48 83 ef 01 55 be 30 02 00 00 48 c1 ef 09 48 89 e5 <48> 8b 3c fd e0 2c bc a0 e8 12 e3 48 ff 5d c3 48 83 ef 01 55 be 
[  858.551505] RIP  [<ffffffffa0b85801>] zio_buf_alloc+0x11/0x20 [zfs]
[  858.551519]  RSP <ffff8800a1dcfa38>
[  858.551521] CR2: ffffffffa0c3ed48
[  858.551525] ---[ end trace 62492988f177682b ]---
[  858.551776] BUG: unable to handle kernel paging request at ffffffffa0c3ed48
[  858.551780] IP: [<ffffffffa0b85801>] zio_buf_alloc+0x11/0x20 [zfs]
[  858.551789] PGD 1e11067 PUD 1e12063 PMD 0 
[  858.551793] Oops: 0000 [#2] PREEMPT SMP 
[  858.551795] Modules linked in: zfs(PO) zunicode(PO) zavl(PO) zcommon(PO) znvpair(PO) spl(O) nvidia(PO) ipv6 iTCO_wdt coretemp crc32c_intel lpc_ich mfd_core
[  858.551804] CPU: 0 PID: 16760 Comm: zpool Tainted: P      D    O  3.15.3 #3
[  858.551806] Hardware name: Gigabyte Technology Co., Ltd. H57M-USB3/H57M-USB3, BIOS F1 01/06/2010
[  858.551808] task: ffff8800cac81b00 ti: ffff8801125ec000 task.ti: ffff8801125ec000
[  858.551809] RIP: 0010:[<ffffffffa0b85801>]  [<ffffffffa0b85801>] zio_buf_alloc+0x11/0x20 [zfs]
[  858.551821] RSP: 0018:ffff8801125ef7b0  EFLAGS: 00010213
[  858.551826] RAX: 000000004b400000 RBX: 0000000001f01c00 RCX: ffffffffa0bbbe20
[  858.551830] RDX: 000000010008864a RSI: 0000000000000230 RDI: 000000000000f80d
[  858.551834] RBP: ffff8801125ef7b0 R08: ffff8801125ec000 R09: 0000000000000000
[  858.551838] R10: 0000000000000047 R11: 0000000000000008 R12: 0000000000000001
[  858.551843] R13: ffff8800593b4568 R14: ffff8800593b45a0 R15: 000000000000f80d
[  858.551848] FS:  00007f7bd61e3b80(0000) GS:ffff880127c00000(0000) knlGS:0000000000000000
[  858.551852] CS:  0010 DS: 0000 ES: 0000 CR0: 000000008005003b
[  858.551857] CR2: ffffffffa0c3ed48 CR3: 00000001125d5000 CR4: 00000000000007f0
[  858.551860] Stack:
[  858.551864]  ffff8801125ef7e8 ffffffffa0af30dd ffff88010d847e70 ffff8800593b4568
[  858.551880]  0000000001f01c00 0000000000000001 000000000000f80d ffff8801125ef818
[  858.551888]  ffffffffa0af5917 ffffc9000bd9ff40 0000000000000000 ffff8801125ef8e4
[  858.551891] Call Trace:
[  858.551897]  [<ffffffffa0af30dd>] arc_space_return+0xbcd/0x3330 [zfs]
[  858.551903]  [<ffffffffa0af5917>] arc_buf_alloc+0xd7/0x100 [zfs]
[  858.551908]  [<ffffffffa0af6112>] arc_read+0x362/0x8e0 [zfs]
[  858.551915]  [<ffffffffa0b8580e>] ? zio_buf_alloc+0x1e/0x20 [zfs]
[  858.551920]  [<ffffffffa0af30ed>] ? arc_space_return+0xbdd/0x3330 [zfs]
[  858.551930]  [<ffffffffa0b0c61d>] dmu_objset_is_receiving+0xbd/0xe30 [zfs]
[  858.551940]  [<ffffffffa0b0c68e>] dmu_objset_is_receiving+0x12e/0xe30 [zfs]
[  858.551950]  [<ffffffffa0b0ce54>] dmu_objset_is_receiving+0x8f4/0xe30 [zfs]
[  858.551960]  [<ffffffffa0b0cd6b>] dmu_objset_is_receiving+0x80b/0xe30 [zfs]
[  858.551970]  [<ffffffffa0b0d6b5>] traverse_pool+0x2d5/0x380 [zfs]
[  858.551981]  [<ffffffffa0b0cf61>] dmu_objset_is_receiving+0xa01/0xe30 [zfs]
[  858.551991]  [<ffffffffa0b0d1d9>] dmu_objset_is_receiving+0xc79/0xe30 [zfs]
[  858.551999]  [<ffffffffa0b818f2>] ? zil_check_log_chain+0x82/0x1b0 [zfs]
[  858.552003]  [<ffffffffa0015493>] ? taskq_init_ent+0x33/0x1f0 [spl]
[  858.552012]  [<ffffffffa0b0d441>] traverse_pool+0x61/0x380 [zfs]
[  858.552023]  [<ffffffffa0b33760>] ? spa_l2cache_drop+0x90/0x1a0 [zfs]
[  858.552036]  [<ffffffffa0b33760>] ? spa_l2cache_drop+0x90/0x1a0 [zfs]
[  858.552045]  [<ffffffffa0b867dc>] ? zio_null+0x5c/0x60 [zfs]
[  858.552059]  [<ffffffffa0b386bc>] spa_lookup_by_guid+0x1bfc/0x2710 [zfs]
[  858.552069]  [<ffffffffa0016421>] ? taskq_create+0x2d1/0x3c0 [spl]
[  858.552082]  [<ffffffffa0b38c48>] spa_lookup_by_guid+0x2188/0x2710 [zfs]
[  858.552087]  [<ffffffffa009a244>] ? zpool_get_rewind_policy+0x194/0x1b0 [zcommon]
[  858.552100]  [<ffffffffa0b3a0f2>] spa_import+0x162/0x5f0 [zfs]
[  858.552110]  [<ffffffffa0b64f2c>] zfs_secpolicy_smb_acl+0x18ec/0x4b60 [zfs]
[  858.552121]  [<ffffffffa0b674d6>] zfs_secpolicy_smb_acl+0x3e96/0x4b60 [zfs]
[  858.552128]  [<ffffffff81084c3d>] ? get_parent_ip+0xd/0x50
[  858.552131]  [<ffffffff818f989d>] ? preempt_count_add+0x4d/0xa0
[  858.552137]  [<ffffffff81173b00>] do_vfs_ioctl+0x2e0/0x4c0
[  858.552141]  [<ffffffff8116be8d>] ? final_putname+0x1d/0x40
[  858.552144]  [<ffffffff8116c104>] ? putname+0x24/0x40
[  858.552150]  [<ffffffff81173d61>] SyS_ioctl+0x81/0xa0
[  858.552153]  [<ffffffff818fd562>] system_call_fastpath+0x16/0x1b
[  858.552157] Code: e8 95 57 fa ff 5b 41 5c 41 5d 41 5e 5d c3 66 66 66 2e 0f 1f 84 00 00 00 00 00 48 83 ef 01 55 be 30 02 00 00 48 c1 ef 09 48 89 e5 <48> 8b 3c fd e0 2c bc a0 e8 12 e3 48 ff 5d c3 48 83 ef 01 55 be 
[  858.552199] RIP  [<ffffffffa0b85801>] zio_buf_alloc+0x11/0x20 [zfs]
[  858.552199]  RSP <ffff8801125ef7b0>
[  858.552201] CR2: ffffffffa0c3ed48
[  858.552203] ---[ end trace 62492988f177682c ]---
[  858.552868] zpool (16760) used greatest stack depth: 11744 bytes left

**Comment by danielhoherd** [comment]
I have experienced this issue on two separate zpools in the last 6 months. Latest was on zfs 0.6.4. neither had ECC memory, both had memory-oversubscribed dedup.

**Comment by behlendorf** [comment]
@danielhoherd thanks for chiming in.  It  would be helpful if you could let is know if you observe this issue with any of the 0.6.5 releases.

**Comment by danielhoherd** [comment]
@behlendorf apologies, I actually did experience this on 0.6.5.x.  The last one I had was a zpool I created in November 2015. My logs don't go back that far so I can't tell what specific version it was, but I am certain it was using the latest available version, so it would certainly have been 0.6.5.x.  Unfortunately I just destroyed that pool so I could recreate it.

If I experience it a third time I'll let you know.
